cmd/object-handlers.go

```diff
@@ -557,8 +624,9 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
 		return
 	}
+
 	if objectAPI.IsEncryptionSupported() {
-		if IsSSECustomerRequest(r.Header) && size > 0 { // handle SSE-C requests
+		if IsSSECustomerRequest(r.Header) && !hasSuffix(object, slashSeparator) { // handle SSE-C requests
```

Comment:
What happens if the object has a `/` suffix but is not zero-size? Actually we are supposed to encrypt... Are non-zero sized objects with `/` handled properly by backend?

Reply:
We ignore such objects.